src/api.v2/helpers/worker/generateEtag.js

```diff
@@ -0,0 +1,42 @@
+const hash = require('object-hash');
+const config = require('../../../config');
+const getExperimentBackendStatus = require('../backendStatus/getExperimentBackendStatus');
+const getExtraDependencies = require('./workSubmit/getExtraDependencies');
+
+const createObjectHash = (object) => hash.MD5(object);
+
+const generateETag = async (
+  data,
+) => {
+  console.log('generateETag', data);
+  const {
+    experimentId,
+    body,
+    extras,
+    cacheUniquenessKey = null,
+  } = data;
+  const backendStatus = await getExperimentBackendStatus(experimentId);
+  const { pipeline: { startDate: qcPipelineStartDate } } = backendStatus;
+
+  const { workerVersion } = config;
+
+  const taskName = body.name;
+  const extraDependencies = await getExtraDependencies(experimentId, taskName);
+
+  const ETagBody = {
+    experimentId,
+    body,
+    qcPipelineStartDate: qcPipelineStartDate.toISOString(),
```

Comment:
When I run this code locally, I get an error on this line. The error I get is: `qcPipelineStartDate.toISOString is not a function`. The type of `qcPipelineStartDate` for me is a string and its value is `2023-10-05T11:51:48.691Z`.

Reply:
I've just run it locally and in staging and work ok 
```
[2023-11-21T09:21:15.153Z] dependencies []
[2023-11-21T09:21:15.153Z] qcPipelineStartDate object 2023-11-21T09:16:36.676Z
[2023-11-21T09:21:15.153Z] ETagBody {
```
Maybe it was old data?

Reply:
this must be out of date then